pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -408,6 +418,32 @@ func (a Adapter) execDevfile(commandsMap common.PushCommandsMap, componentExists
 	return
 }
 
+// TODO: Support Composite
+// execDevfileEvent receives a Devfile Event (PostStart, PreStop etc.) and loops through them
+// Each Devfile Command associated with the given event is retrieved, and executed in the container specified
+// in the command
+func (a Adapter) execDevfileEvent(events []string, compInfo common.ComponentInfo) error {
```

Comment:
Yes! I had intended on doing this initially :) Thanks for pointing this out.

Reply:
I've had a little play around and was reminded of why I ended up choosing against having a single function for both cases.
There is a *slight* difference between the docker functionality, and the Kubernetes functionality in how they define the `ContainerName` in `compInfo`:
- The docker flow requires you to first iterate over the commands, before doing a `GetContainerIDForAlias(containers, command.Exec.Component)` in order to determine the name of the container.
- The Kubernetes flow simply assigns `command.Exec.Component` to the `ContainerName`.

The best solution we could think of was arguably worse than the current implementation.
The function header would have to be something along the lines of:

```
ExecuteDevfileEvent(client ExecClient, data data.DevfileData, events []string, containers []types.Container, isDocker bool, compInfo adaptersCommon.ComponentInfo, logger machineoutput.MachineEventLoggingClient) error 
```
Because we would no longer have access to the adapter we'd pass the `Client`, `DevfileData`, and `MachineEventLoggingClient`. And then we would have to pass both an array of `containers`, and a `ComponentInfo`, because one would be required for docker and the other Kubernetes. And potentially a `bool` to define whether it's docker vs Kube so we could do something like this:
```
if isDocker {
	compInfo.ContainerName = utils.GetContainerIDForAlias(containers, command.Exec.Component)
} else {
	compInfo.ContainerName = command.Exec.Component
}
```
Do you have any suggestions that could solve this without stepping out of the parameters of the spec for this issue? Or which do you think is more suitable - the current solution? Or this singular function? Thanks! :)

Reply:
@maysunfaisal any suggestions since you're most familiar with the exec code?

@upLukeWeston I'm flexible either way, it may be more straightforward to leave as-is, unless Maysun has any suggestions.